### PR TITLE
#776 Allow numpy, np, math in various places we eval user input

### PR DESCRIPTION
**Pull request by rmjarvis**
We had allowed these in config Eval statements already, but @esheldon pointed out in Issue #776 that we don't allow (all of) them in other places like RandomDistribution, where we evaluate user input.

This PR makes sure that all three are allowed in (I think) all such places we use `eval` in GalSim.  It also adds unit tests for this, which is good, since we hadn't had any unit tests for some of this functionality yet (e.g. the Eval type in config).

### CODE REVIEW

**Comment by rmandelb** [comment]
I got errors for all unit tests on this branch (i.e., just EEEEEEEE and so on) with this complaint:
```
ERROR: Failure: SyntaxError (unqualified exec is not allowed in function '_initialize_spec' it contains a nested function with free variables (sed.py, line 135))
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Library/Frameworks/EPD64.framework/Versions/7.2/lib/python2.7/site-packages/nose/loader.py", line 390, in loadTestsFromName
    addr.filename, addr.module)
  File "/Library/Frameworks/EPD64.framework/Versions/7.2/lib/python2.7/site-packages/nose/importer.py", line 39, in importFromPath
    return self.importFromDir(dir_path, fqname)
  File "/Library/Frameworks/EPD64.framework/Versions/7.2/lib/python2.7/site-packages/nose/importer.py", line 86, in importFromDir
    mod = load_module(part_fqname, fh, filename, desc)
  File "/Users/rmandelb/git/GalSim/tests/galsim_test_helpers.py", line 25, in <module>
    import galsim
  File "/Library/Frameworks/EPD64.framework/Versions/7.2/lib/python2.7/site-packages/galsim/__init__.py", line 133, in <module>
    from .sed import SED
SyntaxError: unqualified exec is not allowed in function '_initialize_spec' it contains a nested function with free variables (sed.py, line 135)
```

**Comment by rmjarvis** [comment]
Sorry, it seems that some systems don't like that for some reason.  I was able to reproduce the error on one of my linux machines.  I fixed it there, so please see if it works for you now.

**Comment by rmandelb** [comment]
Yes, it's all good on my Mac right now.  Will test the linux machine + go over the code in the next few days.

File: galsim/bandpass.py [hunk 1]
```diff
@@ -202,12 +202,14 @@ def _initialize_tp(self):
...
+                    self._tp = galsim.utilities.math_eval('lambda wave : ' + self._orig_tp)
                     self._tp(test_wave)
-                except:
+                except Exception as e:
                     raise ValueError(
                         "String throughput must either be a valid filename or something that "+
-                        "can eval to a function of wave. Input provided: {0}".format(self._orig_tp))
+                        "can eval to a function of wave.\n" +
+                        "Input provided: {0}\n".format(self._orig_tp) +
+                        "Caught error: {0}".format(e))
```
Comment (rmandelb) [inline galsim/bandpass.py]: Good idea to add the text of the exception - very helpful.

**Comment by rmandelb** [comment]
Now that all my processes on the Linux machine are finished, I was able to confirm that the unit tests pass there too.  Code looks good to me, as well.  Thanks for this.

**Comment by rmjarvis** [comment]
Thanks Rachel.  I'm going to go ahead and merge this, since I'd like to use this feature in the demo script on branch #782.  If anyone else wanted to comment, please feel free.   I can make the relevant edits on that branch.

**Pull request merged**